Stack/Opc.Ua.Core/Stack/Tcp/TcpTransportListener.cs

```diff
@@ -618,15 +618,12 @@ private void DetectInactiveChannels(object state = null)
         {
             List<TcpListenerChannel> channels;
 
-            lock (m_lock)
+            channels = new List<TcpListenerChannel>();
+            foreach (var chEntry in m_channels)
```

Comment:
Is foreach loop thread safe iterator for concurrent dictionary?

Reply:
From the doc:

> The enumerator returned from the dictionary is safe to use concurrently with reads and writes to the dictionary, however it does not represent a moment-in-time snapshot of the dictionary. The contents exposed through the enumerator may contain modifications made to the dictionary after [GetEnumerator](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getenumerator?view=net-8.0) was called.

[Reference](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getenumerator?view=net-8.0&redirectedfrom=MSDN#remarks)

Reply:
yeah, thats what I figured too. I wasn't sure if an enumerator of just the values is safe, thats why I chose the iterator across the key value pairs.